tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/ConditionsOfJoiningServiceImpl.java

```diff
@@ -43,16 +43,33 @@ public ConditionsOfJoiningServiceImpl(ConditionsOfJoiningRepository repository,
   }
 
   @Override
-  public ConditionsOfJoiningDto save(Long programmeMembershipId, ConditionsOfJoiningDto dto) {
-    LOG.info("Request received to save Conditions of Joining for Programme Membership {}.",
-        programmeMembershipId);
-
-    ProgrammeMembershipDTO programmeMembership = programmeMembershipService.findOne(
-        programmeMembershipId);
+  public ConditionsOfJoiningDto save(Object id, ConditionsOfJoiningDto dto) {
+    UUID realProgrammeMembershipId = null;
+    Long curriculumMembershipId = null;
+    ProgrammeMembershipDTO programmeMembership;
+    try {
+      realProgrammeMembershipId = UUID.fromString(id.toString());
+    } catch (IllegalArgumentException e) {
+      try {
+        curriculumMembershipId = (Long) id;
```

Comment:
My assumption would be that it's better to `Long.parseLong()` the `id.toString()` in case the id has been mapped as an Integer or other compatible number?